src/motion_input_shaping/shaped_lockstep.py

```diff
@@ -0,0 +1,549 @@
+        self.reset_max_speed_limit()
+
+    @property
+    def shaper(self) -> ZeroVibrationShaper | ZeroVibrationStreamGenerator:
+        """Gets the shaper for the specified mode."""
+        match self._shaper_mode:
+            case ShaperMode.DECEL:
+                return self._shaper_decel
+            case ShaperMode.STREAM:
+                return self._shaper_stream
+
+    @property
+    def resonant_frequency(self) -> float:
+        """Get the target resonant frequency for input shaping in Hz."""
+        return self.shaper.resonant_frequency
+
+    @resonant_frequency.setter
+    def resonant_frequency(self, value: float) -> None:
+        """Set the target resonant frequency for input shaping in Hz."""
+        self.shaper.resonant_frequency = value
+
+    @property
+    def damping_ratio(self) -> float:
+        """Get the target damping ratio for input shaping."""
+        return self.shaper.damping_ratio
+
+    @damping_ratio.setter
+    def damping_ratio(self, value: float) -> None:
+        """Set the target damping ratio for input shaping."""
+        self.shaper.damping_ratio = value
+
+    def is_homed(self) -> bool:
+        """Check if all axes in lockstep group are homed."""
+        for axis in self.axes:
+            if not axis.is_homed():
+                return False
+        return True
+
+    def get_setting_from_lockstep_axes(
+        self, setting: str, unit: Units = Units.NATIVE
+    ) -> list[float]:
+        """
+        Get setting values from axes in the lockstep group.
+
+        :param setting: The name of setting
+        :param unit: The values will be returned in these units.
+        :return: A list of setting values
+        """
+        values = []
+        for axis in self.axes:
+            values.append(axis.settings.get(setting, unit))
+        return values
+
+    def set_lockstep_axes_setting(
+        self, setting: str, values: list[float], unit: Units = Units.NATIVE
+    ) -> None:
+        """
+        Set settings for all axes in the lockstep group.
+
+        :param setting: The name of setting
+        :param values: A list of values to apply as setting for each axis or a single value to
+        apply to all
+        :param unit: The values will be returned in these units.
+        """
+        if len(values) > 1:
+            if len(values) != len(self.axes):
+                raise ValueError(
+                    "Length of setting values does not match the number of axes. "
+                    "The list must either be a single value or match the number of axes."
+                )
+            for n, axis in enumerate(self.axes):
+                axis.settings.set(setting, values[n], unit)
+        else:
+            for n, axis in enumerate(self.axes):
+                axis.settings.set(setting, values[0], unit)
+
+    def get_axis_positions(self, unit: Units = Units.NATIVE) -> list[float]:
```

Comment:
Do we need this? This functionality is essentially built into ZML. https://software.zaber.com/motion-library/api/py/ascii/lockstep#_getposition. I'm assuming you've done it for redundancy but I don't think it's necessary.

Reply:
I use this to check the position of each axis in the group and then compare to the limit.max or min for each axis to determine the largest move possible (in the move_max_shaped and move_min_shaped methods). The lockstep get position in ZML gets the position of the primary axis but that might not be the limiting axis.